File: crates/bitwarden/src/sdk/auth/request/two_factor_request.rs
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};
use serde_repr::{Deserialize_repr, Serialize_repr};

#[derive(Serialize_repr, Deserialize_repr, PartialEq, Debug, JsonSchema, Clone)]
#[repr(u8)]
pub enum TwoFactorProvider {
    Authenticator = 0,
    Email = 1,
    Duo = 2,
    Yubikey = 3,
    U2f = 4,
    Remember = 5,
    OrganizationDuo = 6,
    WebAuthn = 7,
}

#[derive(Serialize, Deserialize, Debug, JsonSchema)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct TwoFactorRequest {
    /// Two-factor Token
    pub token: String,
    /// Two-factor provider
    pub provider: TwoFactorProvider,
    /// Two-factor remember
    pub remember: bool,
}

#[derive(Serialize, Deserialize, Debug, JsonSchema)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct TwoFactorEmailRequest {
    /// User Password
    pub password: String,
    /// User email
    pub email: String,
}

File: languages/php/src/schemas/TwoFactorRequest.php
<?php
/**
 * @file ATTENTION!!! The code below was carefully crafted by a mean machine.
 * Please consider to NOT put any emotional human-generated modifications as the splendid AI will throw them away with no mercy.
 */

namespace Bitwarden\Sdk\Schemas;

use Swaggest\JsonSchema\Constraint\Properties;
use Swaggest\JsonSchema\Schema;
use Swaggest\JsonSchema\Structure\ClassStructure;


/**
 * Built from #/definitions/TwoFactorRequest
 */
class TwoFactorRequest extends ClassStructure
{
    const AUTHENTICATOR = 'Authenticator';

    const EMAIL = 'Email';

    const DUO = 'Duo';

    const YUBIKEY = 'Yubikey';

    const U2F = 'U2f';

    const REMEMBER = 'Remember';

    const ORGANIZATION_DUO = 'OrganizationDuo';

    const WEB_AUTHN = 'WebAuthn';

    /** @var string Two-factor Token */
    public $token;

    /** @var string Two-factor provider */
    public $provider;

    /** @var bool Two-factor remember */
    public $remember;

    /**
     * @param Properties|static $properties
     * @param Schema $ownerSchema
     */
    public static function setUpProperties($properties, Schema $ownerSchema)
    {
        $properties->token = Schema::string();
        $properties->token->description = "Two-factor Token";
        $properties->provider = new Schema();
        $propertiesProviderAllOf0 = Schema::string();
        $propertiesProviderAllOf0->enum = array(
            self::AUTHENTICATOR,
            self::EMAIL,
            self::DUO,
            self::YUBIKEY,
            self::U2F,
            self::REMEMBER,
            self::ORGANIZATION_DUO,
            self::WEB_AUTHN,
        );
        $propertiesProviderAllOf0->setFromRef('#/definitions/TwoFactorProvider');
        $properties->provider->allOf[0] = $propertiesProviderAllOf0;
        $properties->provider->description = "Two-factor provider";
        $properties->remember = Schema::boolean();
        $properties->remember->description = "Two-factor remember";
        $ownerSchema->type = Schema::OBJECT;
        $ownerSchema->additionalProperties = false;
        $ownerSchema->required = array(
            self::names()->provider,
            self::names()->remember,
            self::names()->token,
        );
        $ownerSchema->setFromRef('#/definitions/TwoFactorRequest');
    }
}
